cpp/src/plasma/malloc.cc

```diff
@@ -135,10 +135,10 @@ void* fake_mmap(size_t size) {
   int fd = create_buffer(size);
   ARROW_CHECK(fd >= 0) << "Failed to create buffer during mmap";
 #ifdef __linux__
```

Comment:
Let's get rid of this ifdef now; we can still have the comment that explains why we don't use mmap.

Reply:
good idea, done